### PR TITLE
feat: Store integration secret in kuberentes secret (self containment)

### PR DESCRIPTION
**Pull request by alexlokshin-czi**

### CODE REVIEW

File: pkg/backend/aws/backend.go [hunk 1]
```diff
@@ -189,6 +188,23 @@ func NewAWSBackend(
...
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to connect to k8s backend")
+		}
+	} else {
+		computeBackend, err = NewECSComputeBackend(ctx, happyConfig, b)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to connect to ecs backend")
+		}
+	}
+	return computeBackend, nil
```
Comment (jakeyheath) [inline pkg/backend/aws/backend.go:205]:
```suggestion
if happyConfig.TaskLaunchType() == config.LaunchTypeK8S {
	return NewK8SComputeBackend(ctx, happyConfig, b)
}

return NewECSComputeBackend(ctx, happyConfig, b)
```

simplify your nested blocks and if/elses

https://github.com/uber-go/guide/blob/master/style.md#unnecessary-else
https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting

File: pkg/backend/aws/k8s_compute_backend.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	HappyConfig *config.HappyConfig
+}
+
+func NewK8SComputeBackend(ctx context.Context, happyConfig *config.HappyConfig, b *Backend) (interfaces.ComputeBackend, error) {
+	var rawConfig *rest.Config
+	if happyConfig.K8SConfig().AuthMethod == "eks" {
+		// Constructs client configuration dynamically
+		clusterId := happyConfig.K8SConfig().ClusterID
+
+		clusterInfo, err := b.eksclient.DescribeCluster(context.TODO(), &eks.DescribeClusterInput{
```
Comment (jakeyheath) [inline pkg/backend/aws/k8s_compute_backend.go:37]:
```suggestion
		clusterInfo, err := b.eksclient.DescribeCluster(ctx, &eks.DescribeClusterInput{
```

File: pkg/backend/aws/k8s_compute_backend.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+			Name: &clusterId,
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to get k8s cluster configuration")
+		}
+		logrus.Infof("EKS Authenticated K8S Cluster: %s (%s)\n", *(clusterInfo.Cluster).Name, *(clusterInfo.Cluster).Version)
```
Comment (jakeyheath) [inline pkg/backend/aws/k8s_compute_backend.go:43]: I think these types of logs should be debug so they don't show up in the normal flow of the CLI. And if you want to see the more verbose logs, use `-v`. This log is more for us as developers of happy and not our customers.

File: pkg/backend/aws/k8s_compute_backend.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+		cert, _ := base64.RawStdEncoding.DecodeString(*clusterInfo.Cluster.CertificateAuthority.Data)
```
Comment (jakeyheath) [inline pkg/backend/aws/k8s_compute_backend.go:45]: check if not nil and check err

File: pkg/backend/aws/k8s_compute_backend.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+type K8SComputeBackend struct {
+	Backend     *Backend
+	ClientSet   *kubernetes.Clientset
+	HappyConfig *config.HappyConfig
+}
+
+func NewK8SComputeBackend(ctx context.Context, happyConfig *config.HappyConfig, b *Backend) (interfaces.ComputeBackend, error) {
+	var rawConfig *rest.Config
+	if happyConfig.K8SConfig().AuthMethod == "eks" {
```
Comment (jakeyheath) [inline pkg/backend/aws/k8s_compute_backend.go:33]: this auth method is for is they don't have their kubeconfig set up?

File: pkg/backend/aws/k8s_compute_backend.go [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+		Backend:     b,
+		ClientSet:   clientset,
+		HappyConfig: happyConfig,
+	}, nil
+}
+
+func getAuthToken(ctx context.Context, config aws.Config, clusterName string) string {
+	stsClient := sts.NewFromConfig(config)
+	stsSigner := sts.NewPresignClient(stsClient)
+	presignedURLRequest, _ := stsSigner.PresignGetCallerIdentity(ctx, &sts.GetCallerIdentityInput{}, func(presignOptions *sts.PresignOptions) {
```
Comment (jakeyheath) [inline pkg/backend/aws/k8s_compute_backend.go:96]: check your err

**Review by jakeyheath — APPROVED** [review]
Either in a new PR or this one, can we start to build out the unit tests for this k8s module. I think if we wait too long in the development process, we will never do it. We should set up the precedent

**Comment by alexlokshin-czi** [comment]
> Either in a new PR or this one, can we start to build out the unit tests for this k8s module. I think if we wait too long in the development process, we will never do it. We should set up the precedent 

Will do in the next PR.

**Pull request merged**